src/Helper/Helper.php

```diff
@@ -331,4 +378,19 @@ private function getAllRelations(): array {
     return $query->execute()->fetchAllAssoc('job_id');
   }
 
+  /**
+   * Delete job from advanced queue table.
+   *
+   * @param string $jobId
+   *   The advanced queue job id.
+   */
+  private function removeQueueSubmissionRelation(string $jobId): void {
```

Comment:
Use a submission ID to remove stuff. It seems weird to use a job ID since we want to clean up when a submission is deleted (it works with the job ID, but it's easier to understand this if we use the submission ID).